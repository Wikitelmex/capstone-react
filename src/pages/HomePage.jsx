import React from 'react';
import { NavLink } from 'react-router-dom';

const HomePage = () => {
  const title = 'Home Page';
  const reg1 = 'chihuahua';
  const myRoute = `/region/${reg1}`;
  return (
    <div>
      <h1>{title}</h1>
      <p>
        <NavLink to="/Region">Region</NavLink>
      </p>
      <p>
        <NavLink to={myRoute}>
          Region
          {reg1}
        </NavLink>
      </p>
    </div>
  );
};

export default HomePage;
