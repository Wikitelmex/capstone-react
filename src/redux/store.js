import { createStore, applyMiddleware, combineReducers } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

import regionsReducer from './regions/regionsReducer';
import countriesReducer from './countries/countriesReducer';

const mixReducers = combineReducers({
  regions: regionsReducer,
  countries: countriesReducer,
});

const store = createStore(
  mixReducers,
  applyMiddleware(thunk, logger),
);

export default store;
