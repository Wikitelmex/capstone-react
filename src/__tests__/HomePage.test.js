import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import HomePage from '../Pages/HomePage.js';

import store from '../redux/store';

afterEach(cleanup);

it('renders HomePage page with redux', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Router>
        <HomePage />
      </Router>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
