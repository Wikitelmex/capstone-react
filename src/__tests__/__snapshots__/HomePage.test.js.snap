// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomePage page with redux 1`] = `
<DocumentFragment>
  <div>
    <div
      class="d-flex flex-column w-100 h-100 text-center align-items-center justify-content-center"
    >
      <br />
      <br />
      <h1>
        Loading...
      </h1>
      <div
        class="spinner-border"
        role="status"
      >
        <span
          class="visually-hidden"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
